### PR TITLE
Search filesystem for libraries

### PR DESCRIPTION
**Pull request by eholk**
Removes the hardcoded list of libraries from before and instead searches a few library paths.

To do this, I added a `%file-exists?` runtime function, and `string-append` to do the path computations. Adding these required another compiler snapshot.

### CODE REVIEW

**Comment by eholk** [comment]
Had to force push because of snapshot updates from #98. (There really needs to be a better way to manage snapshot changes.)

File: rt/rt.mjs [hunk 1]
```diff
@@ -148,6 +148,8 @@ function rt(engine) {
         '%string->symbol': value => 'S' + value,
         '%symbol->string': x => x.substring(1),
 
+        '%string-append': (x, y) => 's' + x.substring(1) + y.substring(1),
```
Comment (wingo) [inline rt/rt.mjs]: I would think we'd need the same `String()` hack here

File: schism/compiler.ss [hunk 2]
```diff
@@ -178,28 +181,40 @@
...
+                          "./scheme-lib"))
+          (path-suffix (string-append
+                        (fold-left (lambda (path part)
+                                     (string-append
+                                      path
+                                      (string-append "/" (symbol->string part))))
+                                   ""
+                                   name)
+                        ".ss")))
+      (fold-left (lambda (found base-path)
```
Comment (wingo) [inline schism/compiler.ss]: Here it might be nice to define an `or-map`; YMMV though :)

File: rt/rt.mjs (hunk 1 above)
Comment (eholk) [reply rt/rt.mjs]: Good catch. Done.

File: schism/compiler.ss (hunk 2 above)
Comment (eholk) [reply schism/compiler.ss]: Good idea! I think I'll save this for a future cleanup though.

**Pull request merged**